### PR TITLE
dev-python/flask-paginate: verbump to 0.5.3

### PR DESCRIPTION
**Pull request by stkw0**
Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: David Roman <davidroman96@gmail.com>

### CODE REVIEW

File: dev-python/flask-paginate/flask-paginate-0.5.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+DESCRIPTION="Pagination support for flask"
+HOMEPAGE="https://flask-paginate.readthedocs.io"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]"
```
Comment (sbraz) [inline dev-python/flask-paginate/flask-paginate-0.5.3.ebuild]: Can you put everything on one line?

File: dev-python/flask-paginate/flask-paginate-0.5.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{5,6} )
```
Comment (sbraz) [inline dev-python/flask-paginate/flask-paginate-0.5.3.ebuild]: Can you add Python 3.7 support?

**Comment by sbraz** [comment]
Can you squash both commits and force-push?

**Comment by stkw0** [comment]
> Can you squash both commits and force-push?

yep, I'm just fixing what your commented and then I will rebase them

File: dev-python/flask-paginate/flask-paginate-0.5.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]"
+
+RDEPEND="
+	${DEPEND}
+	dev-python/flask[${PYTHON_USEDEP}]"
+
+python_install_all() {
+	distutils-r1_python_install_all
```
[the remark this answers is not in the document]
Comment (stkw0) [reply dev-python/flask-paginate/flask-paginate-0.5.3.ebuild]: The pypi version don't have the tests, and github repository don't have any version tag to fetch from.

**Comment by stkw0** [comment]
> Can you squash both commits and force-push?

Done. Thanks for taking the time to review this :)

File: dev-python/flask-paginate/flask-paginate-0.5.3.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbraz) [reply dev-python/flask-paginate/flask-paginate-0.5.3.ebuild]: I'd open an upstream issue requesting them to tag releases from now on as well as a PR [like this one](https://github.com/rbrcsk/pushbullet.py/pull/104/files) for the PyPI part.

And I'd package the latest commit (510ad833106134711868653fb597bf75ea8ac34f) a bit like I did in [this ebuild](https://github.com/gentoo/gentoo/blob/65f06226327db809b0bec2706ed9be0b86374576/net-analyzer/nagios-plugin-check_lm_sensors/nagios-plugin-check_lm_sensors-4.1.1_p20170912.ebuild).

**Comment by sbraz** [comment]
This latest commit doesn't include tests and `${S}` is not properly set, try something like that:
```bash
# Copyright 1999-2019 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

PYTHON_COMPAT=( pypy{,3} python{2_7,3_{5,6,7}} )

inherit distutils-r1

MY_COMMIT="510ad833106134711868653fb597bf75ea8ac34f"

DESCRIPTION="Pagination support for flask"
HOMEPAGE="https://flask-paginate.readthedocs.io"
# https://github.com/lixxu/flask-paginate/issues/68
SRC_URI="https://github.com/lixxu/flask-paginate/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"

LICENSE="BSD-2"
SLOT="0"
KEYWORDS="~amd64 ~x86"
IUSE="test"

RDEPEND="dev-python/flask[${PYTHON_USEDEP}]"
BDEPEND="
        dev-python/setuptools[${PYTHON_USEDEP}]
        test? ( dev-python/pytest[${PYTHON_USEDEP}] )
"

S="${WORKDIR}/${PN}-${MY_COMMIT}"

python_test() {
        pytest -vv tests/tests.py || die "tests failed with ${EPYTHON}"
}
```

Also it's apparently a 3-clause BSD license, not sure which of ours match it.

**Comment by stkw0** [comment]
> This latest commit doesn't include tests and `${S}` is not properly set, try something like that:
> 
> ```shell
> # Copyright 1999-2019 Gentoo Authors
> # Distributed under the terms of the GNU General Public License v2
> 
> EAPI=7
> 
> PYTHON_COMPAT=( pypy{,3} python{2_7,3_{5,6,7}} )
> 
> inherit distutils-r1
> 
> MY_COMMIT="510ad833106134711868653fb597bf75ea8ac34f"
> 
> DESCRIPTION="Pagination support for flask"
> HOMEPAGE="https://flask-paginate.readthedocs.io"
> # https://github.com/lixxu/flask-paginate/issues/68
> SRC_URI="https://github.com/lixxu/flask-paginate/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
> 
> LICENSE="BSD-2"
> SLOT="0"
> KEYWORDS="~amd64 ~x86"
> IUSE="test"
> 
> RDEPEND="dev-python/flask[${PYTHON_USEDEP}]"
> BDEPEND="
>         dev-python/setuptools[${PYTHON_USEDEP}]
>         test? ( dev-python/pytest[${PYTHON_USEDEP}] )
> "
> 
> S="${WORKDIR}/${PN}-${MY_COMMIT}"
> 
> python_test() {
>         pytest -vv tests/tests.py || die "tests failed with ${EPYTHON}"
> }
> ```
> 
> Also it's apparently a 3-clause BSD license, not sure which of ours match it.

mmm, I was sure I also added the tests, maybe I did it on a different branch. 

the "BSD" license on portage seems to be the 3-clause BSD license

**Comment by sbraz** [comment]
@stkw0 not sure why you didn't keep PyPy from my comment. You also didn't keep the fixed DEPEND…

**Comment by stkw0** [comment]
> @stkw0 not sure why you didn't keep PyPy from my comment. You also didn't keep the fixed DEPEND…

Because I am idiot and I was in a rush and only checked that tests didn't failed, sorry, I will fix it the next release.

**Comment by sbraz** [comment]
No worries, I'll commit it myself with a revbump :)

**Pull request closed without merging**